cpp/Esri/ArcGISRuntime/Toolkit/PopupViewController.cpp

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+ *  Copyright 2012-2020 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#include "PopupViewController.h"
+
+namespace Esri
+{
+namespace ArcGISRuntime
+{
+namespace Toolkit
+{
+
+PopupViewController::PopupViewController(QObject* parent):
+  QObject(parent)
+{
+}
+
+PopupViewController::~PopupViewController()
+{
+}
+
+PopupManager* PopupViewController::popupManager() const
+{
+  return m_popupManager;
+}
+
+void PopupViewController::setPopupManager(PopupManager* popupManager)
+{
+  if (popupManager == m_popupManager)
+    return;
+
+  if (auto a = attachments())
+    disconnect(a, nullptr, this, nullptr);
+
+  if (auto d = displayFields())
+    disconnect(d, nullptr, this, nullptr);
+
+  m_popupManager = popupManager;
```

Comment:
do you need to release the old `m_popupManager`?

Reply:
At the moment we don't connect to anything in the PopupManager - just children elemenets.

Reply:
I see now you are talking about deletion.

You're quite right. PopupManager needs to be wrapped in a `QPointer` if it is non-owning or it needs to delete itself.

I think in the case of PopupStackView, the PopupManagers are not expected to be deleted as you scroll through the list. So I will wrap in a QPointer to make the non-ownership explicit.